### PR TITLE
SCSS Path fixes

### PR DESCRIPTION
**Pull request by undefinedDarkness**
Please follow this template, if applicable.

## Description
Gives eww's scss parser enough rough info to find files from eww's config dir 

## Usage
```
@import "./colors.scss";
@import "colors.scss";
@use "./colors.scss";
@use "colors.scss";
```
These should be vaild.

## Additional Notes
This could be done more simply by asking grass to read the path directly but that would restrict eww's manipulation of the sass data, atm adding env args, I should note that wont work for imported files I think. Maybe fork the scss parser? maybe an easier solution idk

## Checklist

- [ ] All widgets I've added are correctly documented.
- [ ] The documentation in the `docs/content/main` directory has been adjusted to reflect my changes. - not sure if this needs to be documented.
- [ ] I used `cargo fmt` to automatically format all code before committing - I commited that, hopefully that works

### CODE REVIEW

**Comment by undefinedDarkness** [comment]
My git fork is a mess :disappointed_relieved:

**Comment by elkowar** [comment]
In general, if you want to clean up the commit history, read up on interactive rebase (but make sure to backup your changes beforehand ^^)

I'll try out the changes soon! Regarding manipulating imported files, yea, that's actually kind of an issue i haven't yet considered. It might be worth seeing if grass would be willing to add a way for us to intercept into the import resolution, such that we could apply processing to the imported files manually. That's not directly related to this PR tho, so not a blocker

**Comment by elkowar** [comment]
https://github.com/connorskees/grass/issues/46 Issue i filed on grass, for reference

File: src/util.rs [hunk 1]
```diff
@@ -52,10 +52,12 @@ macro_rules! loop_select {
...
 /// then parse it into css.
-pub fn parse_scss_from_file<P: AsRef<Path>>(path: P) -> Result<String> {
-    let scss_content = replace_env_var_references(std::fs::read_to_string(path)?);
-    grass::from_string(scss_content, &grass::Options::default())
-        .map_err(|err| anyhow!("encountered SCSS parsing error: {:?}", err))
+pub fn parse_scss_from_file(path: &Path) -> Result<String> {
+    let config_dir = path.parent().expect("Given SCSS File has no parent?!");
+    let file_content = replace_env_var_references(std::fs::read_to_string(path).expect("Given SCSS File Doesnt Exist!"));
+    let mut grass_config = grass::Options::default();
+    grass_config = grass_config.load_path(config_dir);
```
Comment (elkowar) [inline src/util.rs:59]:
```suggestion
    let grass_config = grass::Options::default().load_path(config_dir);
```

File: src/util.rs [hunk 2]
```diff
@@ -52,10 +52,12 @@ macro_rules! loop_select {
 
 /// read an scss file, replace all environment variable references within it and
 /// then parse it into css.
-pub fn parse_scss_from_file<P: AsRef<Path>>(path: P) -> Result<String> {
-    let scss_content = replace_env_var_references(std::fs::read_to_string(path)?);
-    grass::from_string(scss_content, &grass::Options::default())
-        .map_err(|err| anyhow!("encountered SCSS parsing error: {:?}", err))
+pub fn parse_scss_from_file(path: &Path) -> Result<String> {
+    let config_dir = path.parent().expect("Given SCSS File has no parent?!");
+    let file_content = replace_env_var_references(std::fs::read_to_string(path).expect("Given SCSS File Doesnt Exist!"));
```
Comment (elkowar) [inline src/util.rs:57]:
```suggestion
    let config_dir = path.parent().context("given scss file does not have a parent directory")?;
    let scss_file_content =
        std::fs::read_to_string(path).with_context(|| format!("Given SCSS File Doesnt Exist! {}", path.display()))?;
    let file_content = replace_env_var_references(scss_file_content);
```

File: src/server.rs [hunk 3]
```diff
@@ -26,6 +26,7 @@ pub fn initialize_server(config_dir_override: Option<std::path::PathBuf>) -> Res
         .parent()
         .context("config file did not have a parent?!")?
         .to_owned();
+    std::env::set_current_dir(&config_dir).expect("Failed to change working directory.");
```
Comment (elkowar) [inline src/server.rs:29]:
```suggestion
    std::env::set_current_dir(&config_dir)
        .with_context(|| format!("Failed to set current working directory to {}", config_dir.display()))?;
```

**Review by elkowar — CHANGES_REQUESTED** [review]
Some slight changes, mainly error handling.
Errors shouldn't crash the whole application, but just be passed upwards. Thus, instead of `.expect` (which panics), use `.context` or `.with_context` (provided by `anyhow`) and ? - as I've done in the requested changes. Apart from the requests, everything looks good 👍🏻

**Comment by elkowar** [comment]
Thanks!

**Pull request merged**